app/javascript/src/component_expander.js

```diff
@@ -47,80 +64,101 @@ const {
 **/
 class Expander {
   #config;
+  #state;
 
   constructor($node, config) {
     const conf = mergeObjects({
-      title_tag: 'h2',
-      wrap_content: false,
+      title: null,
+      wrap_content: true,
       auto_open: false,
       duration: 0,
     }, config);
 
+    if(!conf.title) {
+      console.warn('Cannot initialise an Expander without a title element.');
+      return;
+    }
+
     $node.addClass("Expander");
     $node.data("instance", this);
 
     this.#config = conf;
+    this.#state = 'open';
     this.$node = $node;
 
     const id = uniqueString("Expander_");
 
-    const $activator = this.#createActivator(id);
-    const $title = this.$node.find('> ' + conf.title_tag).first();
-    $title.wrapInner($activator);
-    $title.addClass("Expander__title");
-        
+    if(typeof conf.title == 'string') {
+      // We create a button using the title for a label and prepend it to the 
+      // container to toggle disclosure
+      let $activator = this.#createActivator(conf.title);
+      this.#enhanceActivator($activator, id);
+      $node.prepend($activator);
+      this.$activator = $activator;
+    } else if (conf.title.is('button')) {
+      // we enhance the button with the required aria attributes and event
+      // listener
+      this.#enhanceActivator(conf.title, id);
+      this.$activator = conf.title;
+    } else {
+      // We enhance the title element by wrapping its contents with a button
+      let $title = conf.title;
+      let $activator = this.#createActivator();
+      this.#enhanceActivator($activator, id);
+      $title.wrapInner($activator);
+      $title.addClass("Expander__title");
+      this.$activator = $title.find('button');
+    }
```

Comment:
All three cases are calling `this.#enhanceActivator()` which has me believe the code should be inside that function.

I'm also wondering if (in the first condition when a string), `$node.prepend($activator);` will work. 
Isn't `$node` the thing that opens and shuts, so the $activator will be inside and become invisible?

Reply:
Yep, I wasn't super happy with this.  The problem is that all 3 cases do something slightly different... 

Refactored to have `#createActivator` call `#enhanceActivator` as we're never not going to want to do that.  So we always get an 'enhanced' activator back from `#createActivator` but we still have a spearate enhance function to call if we're passed a `<button>`

`$node` is the whole containing element, not just the expanding/contracting thing. Structure is:
```
<$node>
  <$activator />
  <$container>
    content
  </$container>
</$node>
```
I've manually tested all versions of the setup so it definitely works. (And will write some JS tests for it too)